File: src/darkyenus/resourcepacker/tasks/IgnoreTask.scala
package darkyenus.resourcepacker.tasks

import com.esotericsoftware.minlog.Log
import darkyenus.resourcepacker.{ResourceFile, ResourceDirectory, Task}

/**
 * Removes all files and directories marked with `.ignore.`
 * @author Darkyen
 */
object IgnoreTask extends Task {

  override def operate(file: ResourceFile): Boolean = {
    if(file.flags.contains("ignore")){
      file.removeFromParent()
      Log.info(Name, "File ignored. (" + file + ")")
      true
    }else false
  }

  override def operate(directory: ResourceDirectory): Boolean = {
    if(directory.flags.contains("ignore")){
      directory.parent.removeChild(directory)
      Log.info(Name, "Directory ignored. (" + directory + ")")
      true
    }else false
  }
}

File: src/main/java/com/darkyen/resourcepacker/tasks/IgnoreTask.kt
package com.darkyen.resourcepacker.tasks

import com.darkyen.resourcepacker.Resource.ResourceDirectory
import com.darkyen.resourcepacker.Resource.ResourceFile
import com.darkyen.resourcepacker.Task
import com.esotericsoftware.minlog.Log

/**
 * Removes all files and directories marked with `.ignore.`
 * @author Darkyen
 */
object IgnoreTask : Task() {

    override fun operate(file: ResourceFile): Boolean {
        if (file.flags.contains("ignore")) {
            file.removeFromParent()
            Log.info(Name, "File ignored. ($file)")
            return true
        }

        return false
    }

    override fun operate(directory: ResourceDirectory): Boolean {
        if (directory.flags.contains("ignore")) {
            directory.parent.removeChild(directory)
            Log.info(Name, "Directory ignored. ($directory)")
            return true
        }

        return false
    }
}
